File: pat1005.cpp
#include<iostream>
#include<string>
using namespace std;
int main() {
	string s;
	cin >> s;
	int sum = 0;
	for (int i = 0; i < s.length(); i++) {
		sum += s[i] - '0';
	}
	s = to_string(sum);
	for (int i = 0; i < s.length(); i++) {
		switch (s[i])
		{
		case '0': cout << "zero"; break;
		case '1': cout << "one"; break;
		case '2': cout << "two"; break;
		case '3': cout << "three"; break;
		case '4': cout << "four"; break;
		case '5': cout << "five"; break;
		case '6': cout << "six"; break;
		case '7': cout << "seven"; break;
		case '8': cout << "eight"; break;
		case '9': cout << "nine"; break;
		default:
			break;
		}
		if (i != s.length() - 1){
			cout << ' ';
		}
	}

	return 0;
}

File: pat1005.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        String[] numOfCN = {"zero", "one", "two", "three", "four",
                "five", "six", "seven", "eight", "nine"};
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String line = reader.readLine();
        int sumOfDigits = 0;
        for (int i = 0; i < line.length(); i++) {
            sumOfDigits += line.charAt(i) - '0';
        }
        String sumInString = Integer.toString(sumOfDigits);     // 将和转化化字符串形式
        for (int i = 0; i < sumInString.length(); i++) {
            System.out.print(numOfCN[sumInString.charAt(i) - '0']);
            if (i != sumInString.length() - 1) {
                System.out.print(" ");
            }
        }
    }
}
